tests/restricted_rooms_test.go

```diff
@@ -114,6 +114,15 @@ func checkRestrictedRoom(t *testing.T, alice *client.CSAPI, bob *client.CSAPI, a
 		// Wait until Alice sees Bob leave the allowed room. This ensures that Alice's HS
 		// has processed the leave before Bob tries rejoining, so that it rejects his
 		// attempt to join the room.
+		alice.MustSyncUntil(t, client.SyncReq{}, client.SyncTimelineHas(
+			room, func(ev gjson.Result) bool {
+				if ev.Get("type").Str != "m.room.member" || ev.Get("sender").Str != bob.UserID {
+					return false
+				}
+
+				return ev.Get("content").Get("membership").Str == "leave"
+			}))
+
```

Comment:
This is a duplicate of the stuff underneath? Why add this?

Reply:
It makes sure the leave worked in both rooms now